python/test/unit/language/test_core.py

```diff
@@ -3788,13 +3788,17 @@ def _kernel(dst, src, CACHE: tl.constexpr):
 
     if not is_cuda():
         if is_hip():
+            target_arch = get_arch()
```

Comment:
Can we move the `if is_hip()` to be outside and ahead of the `if not is_cuda`?

Reply:
in this case ``if not is_cuda`` won't make any sense

Reply:
It's bypassing the following cuda specific checks for out-of-tree third-party backends right?